include/aws/common/byte_buf.h

```diff
@@ -497,6 +497,32 @@ AWS_STATIC_IMPL bool aws_byte_cursor_read_be64(struct aws_byte_cursor *cur, uint
     return rv;
 }
 
+/**
+ * Appends a sub-buffer to the specified buffer.
+ *
+ * If the buffer has at least `len' bytes remaining (buffer->capacity - buffer->len >= len),
+ * then buffer->len is incremented by len, and an aws_byte_buf is assigned to *output corresponding
+ * to the last len bytes of the input buffer. The aws_byte_buf at *output will have a null
+ * allocator, a zero initial length, and a capacity of 'len'. The function then returns true.
+ *
+ * If there is insufficient space, then this function nulls all fields in *output and returns
+ * false.
+ */
+AWS_STATIC_IMPL bool aws_byte_buf_advance(
```

Comment:
byte_cursor_advance is implemented like:
```c
AWS_STATIC_IMPL struct aws_byte_cursor aws_byte_cursor_advance(struct aws_byte_cursor *cursor, size_t len) {
    struct aws_byte_cursor rv;
    if (cursor->len > (SIZE_MAX >> 1) || len > (SIZE_MAX >> 1) || len > cursor->len) {
        rv.ptr = NULL;
        rv.len = 0;
    } else {
        rv.ptr = cursor->ptr;
        rv.len = len;

        cursor->ptr += len;
        cursor->len -= len;
    }

    return rv;
}
```
I think we should either:
1. Make byte_cursor_advance follow this model of returning a bool (and also make output optional)
2. Make this function conform to the ways of byte_cursor_advance

Reply:
While 1 is a reasonable ask, I'd like to finish fixing our code for all the existing breaking changes before introducing even more. Can we defer that to a later PR?